massa-network/src/network_worker.rs

```diff
@@ -995,36 +975,54 @@ impl NetworkWorker {
-    /// * peer_info_db: Database with peer information.
     /// * cur_connection_id : connection id of the node we are trying to reach
-    /// * active_connections: hashmap linking connection id to ipAddr to whether connection is outgoing (true)
-    /// * event_tx: channel to send network events out.
     async fn manage_in_connections(
         &mut self,
         res: std::io::Result<(ReadHalf, WriteHalf, SocketAddr)>,
         cur_connection_id: &mut ConnectionId,
     ) -> Result<(), NetworkError> {
         match res {
             Ok((reader, writer, remote_addr)) => {
-                if self.peer_info_db.try_new_in_connection(&remote_addr.ip())? {
-                    let connection_id = *cur_connection_id;
-                    debug!(
-                        "inbound connection from addr={} succeeded => connection_id={}",
-                        remote_addr, connection_id
-                    );
-                    massa_trace!("in_connection_established", {
-                        "ip": remote_addr.ip(),
-                        "connection_id": connection_id
-                    });
-                    cur_connection_id.0 += 1;
-                    self.active_connections
-                        .insert(connection_id, (remote_addr.ip(), false));
-                    self.new_connection(connection_id, reader, writer)?;
-                } else {
-                    debug!("inbound connection from addr={} refused", remote_addr);
-                    massa_trace!("in_connection_refused", {"ip": remote_addr.ip()});
+                match self.peer_info_db.try_new_in_connection(&remote_addr.ip()) {
+                    Ok(_) => {
+                        let connection_id = *cur_connection_id;
+                        debug!(
+                            "inbound connection from addr={} succeeded => connection_id={}",
+                            remote_addr, connection_id
+                        );
+                        massa_trace!("in_connection_established", {
+                            "ip": remote_addr.ip(),
+                            "connection_id": connection_id
+                        });
+                        cur_connection_id.0 += 1;
+                        self.active_connections
+                            .insert(connection_id, (remote_addr.ip(), false));
+                        self.manage_successfull_connection(connection_id, reader, writer)?;
+                    }
+                    Err(NetworkError::PeerConnectionError(
+                        NetworkConnectionErrorType::MaxPeersConnectionReached(_),
+                    )) => {
+                        debug!("Maximum connection reached");
+                        debug!("inbound connection from addr={} refused", remote_addr);
+                        let peer_list = self.peer_info_db.get_advertisable_peer_ips();
+                        WriteBinder::new(writer)
```

Comment:
Maybe here add a timeout, to avoid writing for too long if there is an issue

Reply:
Good point

Reply:
Done!